backend/src/api.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+
+from flask import Blueprint, jsonify, request
+from sqlalchemy.orm.exc import NoResultFound
+
+from .model import Transaction, db
+
+api = Blueprint("api", __name__, url_prefix="/api")
+
+
+@api.route("/transactions/<int:transaction_id>", methods=["PUT", "GET", "DELETE"])
+def transaction(transaction_id):
+    transaction = Transaction.query.filter_by(id=transaction_id).first()
+    if not transaction:
+        return jsonify({"result": "failure", "error": 404}), 404
+
+    if request.method == "PUT":
+        if request.is_json:
+            updates = request.json["updates"]
```

Comment:
I think moving logic to dedicated class in routes will be good idea (easier to maintain) 
example:
```
if request.method == "PUT":
   try {
       return jsonify(SomeClass.put_action().serialize), 200 
   } except (someLocalError) {
       return jsonify({"result": "failure", "error": 400}), 400
   }
````